src/ai/ai.cpp

```diff
@@ -22,23 +57,12 @@ void update_ai_general_status(sys::state& state) {
 		for(auto b : state.world.nation_get_nation_adjacency_as_connected_nations(n)) {
 			auto other = b.get_connected_nations(0) != n ? b.get_connected_nations(0) : b.get_connected_nations(1);
 			if(!nations::are_allied(state, n, other) && (!in_sphere_of || in_sphere_of != other.get_in_sphere_of())) {
-				greatest_neighbor = std::max(greatest_neighbor, float(state.world.nation_get_military_score(other)));
+				greatest_neighbor = std::max(greatest_neighbor, estimate_strength(state, other));
 			}
 		}
 
-		float self_str = float(state.world.nation_get_military_score(n));
-		for(auto subj : n.get_overlord_as_ruler()) {
-			self_str += 0.75f * float(subj.get_subject().get_military_score());
```

Comment:
I want to maintain this discount when deciding how secure the nation is, because puppets are somewhat less useful on the defense